### PR TITLE
Docs review for Contribute section

### LINKED ISSUE #1249: [Docs Review] Contribute
Parent Issue vitessio/website#1222 
Vitess github url: https://github.com/vitessio/website

### Pages to review:

- [x] [Contribute](https://vitess.io/docs/contributing/)
- [x] [Build on macOS](https://vitess.io/docs/contributing/build-on-macos/)
- [x] [Build on Ubuntu/Debian](https://vitess.io/docs/contributing/build-on-ubuntu/)
- [x] [Build on CentOS](https://vitess.io/docs/contributing/build-on-centos/)
- [x] [GitHub Workflow](https://vitess.io/docs/contributing/github-workflow/)
- [x] [Coding Standards](https://vitess.io/docs/contributing/code-reviews/)
- [x] [Sample First Issue](https://vitess.io/docs/contributing/sample-first-issue/)
- [x] [Parser and AST](https://vitess.io/docs/contributing/contributing-to-ast-parser/)
- [x] [Release Notes](https://vitess.io/docs/contributing/release-notes/)

### What to do 

- [ ] Make sure the content reflect the current state and facts are up-to-date.
- [ ] Add details if needed either in form of new section on current page or adding new page.
- [ ] Add links/Urls on page are working
- [ ] Deprecate flags should be marked as 'deprecated'
- [ ] Review is done for both V15 and V16

Please note that we are not going to focus of restructuring the website content as part of this project. It is something we will do as a separate project later on.

### PR DESCRIPTION
**Pull request by GuptaManan100**
### Description

This PR does the docs review for the contribute section of vitess

### Related Issues
- Fixes #1249

### CODE REVIEW

File: content/en/docs/contributing/release-notes.md [hunk 1]
```diff
@@ -17,26 +17,18 @@ First, we need to `git fetch` the remote we want to work with and make sure we a
...
-The `SUMMARY` argument is optional, it is a path to a README file that contains a text that will added in the `Announcement` section of the release notes.
-
-### How to add pull requests to the release notes?
-
-When generating the release notes, only the pull requests labeled with `release notes` will be listed.
-
-Pull requests can also be labeled with `release notes (needs details)`.
-Those will be listed in the `Announcement` section of the release note with a special `TODO` that requires human intervention for further explanation on what the pull request do.
-For instance, it can be an important pull request that deprecates a part of Vitess, or an important feature that we want to talk about.
+The `SUMMARY` argument is optional, it is a path to a README file that contains text to prefix the release notes with that the maintainers wrote before the release.
```
Comment (rsajwani) [inline content/en/docs/contributing/release-notes.md]: with that the maintainers
or 
with what the maintainers

File: content/en/docs/contributing/sample-first-issue.md [hunk 2]
```diff
@@ -113,14 +113,16 @@ We can now verify that our added unit test in *token_test.go* works perfectly an
...
+If the changes in the PR are significant enough to warrant adding a section for it in the release notes, then we should do that.
+After that, we need to create a pull request for our changes and address any review comments. 
 While creating the pull request, we need to take care of a few things -
 
 1. Follow the existing template for pull requests
 2. Add a description of what has been fixed
 3. Add the label for the correct component affected by the changes. In our case that is `Component: Query Serving`.
 4. Add the label for the type of changes in the PR. In our case that is `Type: Enhancement`.
-5. Add the label describing whether this PR should be included in the release notes or not. In our case that is `release notes`.
+5. Add the label describing whether this PR should be backported to some of the previous releases as well. The labels look like `Backport to: release-15.0`
```
Comment (rsajwani) [inline content/en/docs/contributing/sample-first-issue.md]: minor: 
previous releases as well. For example in order to back port to release 15.0 we use the label `Backport to: release-15.0`

**Review by rsajwani — APPROVED** [review]

File: content/en/docs/contributing/sample-first-issue.md (hunk 2 above)
Comment (GuptaManan100) [reply content/en/docs/contributing/sample-first-issue.md]: Fixed

File: content/en/docs/contributing/release-notes.md (hunk 1 above)
Comment (GuptaManan100) [reply content/en/docs/contributing/release-notes.md]: Fixed

**Pull request merged**